src/librustc_target/spec/aarch64_apple_darwin.rs

```diff
@@ -0,0 +1,32 @@
+use crate::spec::{LinkerFlavor, Target, TargetOptions, TargetResult};
+
+pub fn target() -> TargetResult {
+    let mut base = super::apple_base::opts();
+    base.cpu = "apple-a12".to_string();
+    base.max_atomic_width = Some(128);
+    base.eliminate_frame_pointer = false;
```

Comment:
This might be interesting to test out removing, targets generally benefit from frame pointer elimination and unless we otherwise really need to this is likely best left as `true`. (basically try leaving it as `true` until proven otherwise)

Reply:
Do you happen to recall why the x86-apple-darwin target has this marked as false (I haven't even done basic git history searching yet, so it might be obvious if I do). I'd have expected that the two platforms would match for this.

Reply:
The oldest git commit I found was
```
commit 6b130e3dd9547e233dca2cfd72a5968891672d9c
Author: Corey Richardson <corey@octayn.net>
Date:   Wed Jul 23 11:56:36 2014 -0700

    Implement flexible target specification
```
That adds the whole file at once, so it's not clear whether this is a rename or when it was first written. `eliminate_frame_pointer = false` is present in this commit.

Reply:
It's _possible_ that we could have always had this be true for x86_64-apple-darwin and might get a little speed bump if we add it now :-)

Reply:
Thanks for the digging @jyn514, the fp elim here comes from the code contents of the patch:

```rust
    // FIXME: #11906: Omitting frame pointers breaks retrieving the value of a parameter.
    // FIXME: #11954: mac64 unwinding may not work with fp elim
    let no_fp_elim = (sess.opts.debuginfo != NoDebugInfo) ||
                     (sess.targ_cfg.os == abi::OsMacos &&
                      sess.targ_cfg.arch == abi::X86_64);
```

which points to https://github.com/rust-lang/rust/issues/11954 which I believe was [incorrectly closed](https://github.com/rust-lang/rust/issues/11954#issuecomment-283374851) (only references i686, not x86_64).

This sounds vaguely familiar about how it's related to unwinding. This also is the definition of something lost to time which we unfortunately lost track of :(.

In any case I would not enable this option for AArch64 unless you're sure that unwinding or something else needs the frame pointer (which everything modern nowadays doesn't need). I believe we should probably try to enable the frame pointer on x86_64 as well, especially given how high-profile that is.

Reply:
Opened https://github.com/rust-lang/rust/pull/74575 and removed it from here.